### PR TITLE
Drupal 11 prototype

### PR DESCRIPTION
**Pull request by jackrabbithanna**
Overview
----------------------------------------
Updates to support Drupal 11.

The main thing is updating composer.json to allow Symfony 7 versions.

Update some deprecated functions

TODO
----------------------------------------
Check if deprecated API changes are backward compatible for all Drupal versions CiviCRM officially supports
Remove temporary condition for composer-compile-lib test branch when this PR is tested and merged: https://github.com/civicrm/composer-compile-lib/pull/6
Other deprecated API to fix?
Testing!

### CODE REVIEW

**Comment by jackrabbithanna** [comment]
Good reference for changes from Symfony 6 to 7 https://github.com/symfony/symfony/blob/7.1/UPGRADE-7.0.md

**Comment by jackrabbithanna** [comment]
A way to setup a local with D11 + CiviCRM https://github.com/jackrabbithanna/drupal11-civicrm-proto
Using my forks' branches.
Installation went smooth, testing is looking good. At least basics are working. Tried installing Mosaico and some other extensions. So far so good! Maybe not a big lift or alot of changes in those bits of Symfony that Civi uses.

**Comment by jackrabbithanna** [comment]
@demeritcowboy I've removed that dev-d11-test reference from the changes in the PR. I'm hoping to keep the one in the D8 repo until this one is merged, so the prototype composer template I had made can keep working for people.

File: CRM/Core/Permission/Drupal8.php [hunk 1]
```diff
@@ -93,11 +93,12 @@ public function permissionEmails($permissionName) {
...
       return $_cache[$permissionName];
     }
-
+    $roles = \Drupal\user\Entity\Role::loadMultiple();
+    unset($roles[\Drupal\user\RoleInterface::ANONYMOUS_ID]);
     $role_ids = array_map(
       function (\Drupal\user\RoleInterface $role) {
         return $role->id();
-      }, user_roles(TRUE, $permissionName)
+      }, array_filter($roles, fn(\Drupal\user\RoleInterface $role) => $role->hasPermission($permissionName))
```
Comment (demeritcowboy) [inline CRM/Core/Permission/Drupal8.php:101]: I checked and this `fn` syntax still works in php 7.4. 👍 

I find I have to watch my effin' syntax or else the effin' computer goes api-shift...

File: composer.json [hunk 2]
```diff
@@ -56,16 +56,16 @@
     "dompdf/dompdf" : "~2.0.4",
     "firebase/php-jwt": ">=3 <7",
     "rubobaquero/phpquery": "^0.9.15",
-    "symfony/config": "~4.4 || ~6.0",
+    "symfony/config": "~4.4 || ~6.0 || ~7.0",
```
Comment (demeritcowboy) [inline composer.json:59]: For symfony/config, Drupal 10 people get either 6 or 7 depending on their php version. The changelog from 6.4 to 7.0 is almost entirely adding typehints and declarations.

Comment (demeritcowboy) [reply composer.json:59]: For the other symfonys, drupal 10 itself limits to 6 so no change.

**Comment by demeritcowboy** [comment]
The composer.lock hash will need updating, but we can do that separately.

**Pull request merged**